main.go

```diff
@@ -590,7 +591,23 @@ func main() {
 	logPth, err := saveRawOutputToLogFile(rawXcodebuildOutput, (testErr == nil))
 
 	if err != nil {
-		log.Warnf("Failed to save the Raw Output, error %s", err)
+		log.Warnf("Failed to save the Raw Output, error: %s", err)
+	}
+
+	// exporting xcresult only if test result dir is present
+	if testResultPath := os.Getenv(bitriseConfigs.BitriseTestResultDirEnvKey); len(testResultPath) > 0 {
+		fmt.Println()
+		log.Infof("Exporting test results")
+
+		// the leading `/` means to copy not the content but the whole dir
+		// -a means a better recursive, with symlinks handling and everything
+		cmd := command.New("cp", "-a", buildTestParams.TestOutputDir, testResultPath+"/")
```

Comment:
what about the attached images? afaik they are in a subdirectory

Reply:
If you use the -resultBundlePath flag, then the xcode will export the files from the .xctestresult and not he .xctestresults itself:

```
❯ tree                                                                                                                                                                                    ~/Desktop/ide_2
.
├── 1_Test
│   ├── Attachments
│   │   ├── Screenshot\ of\ main\ screen\ (ID\ 1)_1_2C61C577-ABE8-42E0-8229-E13AB615A2D6.png
│   │   └── Screenshot\ of\ main\ screen\ (ID\ 1)_1_FD154EA7-F829-4279-9E55-B3590F0F00E9.png
│   ├── Diagnostics
│   │   ├── BullsEyeUITests-4249BD04-771B-48C5-8D15-0D0040FF44C2
│   │   │   ├── BullsEyeUITests-1C28C059-968A-440F-B1A2-98C968020444
│   │   │   │   ├── LaunchSessions
│   │   │   │   │   └── 122017A2-ECCA-46FB-AFA6-95792D604957
│   │   │   │   │       └── remote-container
│   │   │   │   │           └── tmp
│   │   │   │   │               └── BullsEyeUITests-122017A2-ECCA-46FB-AFA6-95792D604957.xctestconfiguration
│   │   │   │   ├── Session-BullsEyeUITests-2019-03-08_100731-j9sfkz.log
│   │   │   │   ├── StandardOutputAndStandardError-com.bitfall.project-with--parallel-ui-test.txt
│   │   │   │   └── StandardOutputAndStandardError.txt
│   │   │   └── scheduling.log
│   │   └── iPhone\ 7\ Plus_0C804187-FB0C-45F2-9E3E-4FF53ED4ECC7
│   │       └── testmanagerd.log
│   ├── action.xcactivitylog
│   ├── action_TestSummaries.plist
│   └── build.xcactivitylog
├── Attachments
│   ├── Screenshot\ of\ main\ screen\ (ID\ 1)_1_2C61C577-ABE8-42E0-8229-E13AB615A2D6.png
│   └── Screenshot\ of\ main\ screen\ (ID\ 1)_1_FD154EA7-F829-4279-9E55-B3590F0F00E9.png
├── Info.plist
└── TestSummaries.plist

11 directories, 15 files
```

Reply:
@BirmacherAkos please link the corresponding logic here.

Reply:
@godrei https://github.com/bitrise-io/steps-deploy-to-bitrise-io/pull/79#issuecomment-470885094

That PR needs to be updated otherwise we will have shared logic ina ll of our xcode steps instead having the logic in a shared place.